HAM/data_owner/assets/dataset/opener.py

```diff
@@ -0,0 +1,139 @@
+        return y
+    
+    def save_predictions(self, y_pred, path):
+        torch.save(y_pred, path)
+        return
+
+    def get_predictions(self, path):
+        # return pd.read_csv(path)
+        return torch.load(path)
+
+    def fake_X(self):
+        raise NotImplementedError
+
+    def fake_y(self):
+        raise NotImplementedError
+    
+    @classmethod
+    def _get_X(cls, data):
+        return 
+    
+    @classmethod
+    def _get_y(cls, data):
+        return data.loc[:, 'dx_idx']
+
+    @classmethod
+    def _fake_data(cls):
+        raise NotImplementedError
+    
+    @classmethod
+    def _get_data(cls, folders):
+        # #find images files
+        paths_image = []
+        paths_csv = []
+        root_folder = os.path.join(*folders[0].split('/')[:-1])
+        for folder in folders:
+            paths_image += [os.path.join(folder, f) for f in os.listdir(folder) if f[-4:] == '.jpg']
+            paths_csv += [os.path.join(folder, f) for f in os.listdir(folder) if f[-4:] == '.csv']
+
+        for path in paths_csv:
+            df_train = pd.read_csv(path)
+            df_train['dx_idx'] = pd.Categorical(df_train['dx']).codes
+        # norm_mean, norm_std = self.calculate_mean_std(df_train['filename'].to_list())
+
+        train_transform = transforms.Compose([transforms.Resize((224, 224)),
+                                            # transforms.RandomHorizontalFlip(),
+                                            # transforms.RandomVerticalFlip(),
+                                            # transforms.RandomRotation(20),
+                                            # transforms.ColorJitter(brightness=0.1, contrast=0.1, hue=0.1),
+                                            transforms.ToTensor(),
+                                            # transforms.Normalize(norm_mean, norm_std)
+                                            ])
+
+        train_dataset = HAM(df_train.reset_index(), train_transform, paths_image)
+        params = {'batch_size': 1,
+          'shuffle': False,
+          'num_workers': 0}
+        train_loader = DataLoader(train_dataset, **params)
+
+        X = []
+        y = []
+        for i, j in train_loader:
+            X.append(i)
+            y.append(j)
+
+        return X, y
+
+    # @classmethod
+    # def calculate_mean_std(cls, paths):
+
+    #     (img_h, img_w) = (224, 224)
+    #     imgs = []
+    #     means, stds = [], []
+
+    #     for i in tqdm(range(len(paths))):
+    #         img = cv2.imread(paths[i])
+    #         img = cv2.resize(img, (img_h, img_w))
+    #         imgs.append(img)
+
+    #     imgs = np.stack(imgs, axis=3)
+    #     imgs = imgs/255.
+
+    #     for i in range(3):
+    #         pixels = imgs[:, :, i, :].ravel()
+    #         means.append(np.mean(pixels))
+    #         stds.append(np.std(pixels))
+
+    #     print("normMean = {}".format(means))
+    #     print("normStd = {}".format(stds))
+
+    #     return means, stds
```

Comment:
Why is this method commented?

Reply:
It was to normalize data but I don't use it in this example finally. I can remove it.